SoundGrid.py

```diff
@@ -0,0 +1,184 @@
+		self.model = model
+		self.screen = pygame.display.set_mode(size)
+		
+	def draw(self):
+		""" Draws the blocks to the pygame window """
+		self.screen.fill(pygame.Color('black'))
+		for column in self.model.blocks:
+			for block in column:
+				r = pygame.Rect(block.left,
+								block.top,
+								block.size,
+								block.size)
+				pygame.draw.rect(self.screen, block.color,r)
+		pygame.display.update()             
+
+
+class sound_grid_model(object):
+	""" Represents the interaction state for the visualization """
+
+	def __init__(self, width, height):
+		self.width = width
+		self.height = height
+
+		self.margin = 5
+		self.block_no = 15
+		self.block_size = (size[0] - self.margin *(self.block_no + 1)) /self.block_no
+		
+		self.block_ranges = []
+		block_range_column = []
+
+		self.blocks = []
+		column = []
+
+		# generates columns of blocks by sweeping x- and y-axes
+		# blocks stored in self.blocks
+		for left in range(self.margin, #beginning of range
+						  self.width - self.margin - self.block_size, #end of range
+						  self.margin + self.block_size): #step size
+			for top in range(self.margin, 
+							 self.height,
+							 self.margin + self.block_size):
+			  	if len(column) <= self.block_no - 1:
+					column.append(block(left, top, self.block_size))
+					block_range_column.append(((left, left + self.block_size), (top, top + self.block_size)))
+			if len(self.blocks) <= self.block_no - 1:
+				self.blocks.append(column)
+				self.block_ranges.append(block_range_column)
+				column = []
+				block_range_column = []
+
+
+class block(object):
+	""" Represents a block in the musical visualization grid """
+
+	def __init__(self, left, top, size, color = dark_grey):
+		self.left = left
+		self.top = top
+		self.size = size
+		self.color = color
+
+
+class controller(object):
+	""" Manages user input """
+	def __init__(self, model):
+		self.model = model
+		self.toggle = [[0 for row in range(self.model.block_no)] for column in range(self.model.block_no)]
+
+		# sound initializations
+		pygame.mixer.set_num_channels(self.model.block_no) # sets number of channels to number of blocks
+		self.sounds = ['Sounds/Note1.aiff',
+						'Sounds/Note2.aiff',
+						'Sounds/Note3.aiff',
+						'Sounds/Note5.aiff',
+						'Sounds/Note6.aiff',
+						'Sounds/Note7.aiff',
+						'Sounds/Note8.aiff',
+						'Sounds/Note10.aiff',
+						'Sounds/Note11.aiff',
+						'Sounds/Note12.aiff',
+						'Sounds/Note13.aiff',
+						'Sounds/Note15.aiff',
+						'Sounds/Note16.aiff',
+						'Sounds/Note17.aiff',
+						'Sounds/Note18.aiff',]
+
+	def handle_event(self, event):
+		""" When a block is clicked, the color of the block changes color and a sound plays
+		    A wave of grey moves out from the block clicked
+
+		    event: pygame event while running"""
+
+		if event.type == QUIT:
+			return
+
+		if event.type == KEYDOWN:
+			if event.key == pygame.K_ESCAPE:
+			   pygame.quit()
+			   return
+
+		for column in range(len(self.model.block_ranges)):
```

Comment:
Since you use `self.model.block_ranges` several times, it might be worth assigning this to a new variable with a shorter name.

Not sure what the text formatting looks like on your end, but in cases where you have long lines, you can use a linebreak `\`  or add parentheses with implicit continuation.